### DISCUSSION TITLE
Geolocation/autocompletion, embrace both :)

### DISCUSSION THREAD

**Discussion by olivier5741** [komoot/photon#706] [Ideas]
Hi there, I would like to use photon for geolocation (works pretty good) but can't get rid of it's autocompletion oriented focus ...

For street layer, 
`Street of nowhere, 1050 Ixelles, Belgique` -> gives 0 result (which is good :) )
`Avenue of nowhere, 1050 Ixelles, Belgique` -> gives me 10 results, which may be good for autocompletion but completely irrelevant for geolocation ... -> and I can't get rid of this ...

https://photon.komoot.io/api?layer=street&q=Avenue%20of%20nowhere,%201050%20Ixelles,%20Belgique

**Comment by lonvia** [comment]
If you run your own instance, you could compile your own version of Photon with the following lines commented out: https://github.com/komoot/photon/blob/716b45f969afa22c72f3ab210515c0a5e3e9fee2/src/main/java/de/komoot/photon/elasticsearch/ElasticsearchSearchHandler.java#L39-L41

This will give you a version of Photon that will insist on exact matches. We could make this configurable but the question is, if you really want that. For example, I imagine that you would still be happy if Nominatim found 'Rue de la Matin' even when you search for 'Rue du Matin'. So what most people really want is to use spelling correction "wisely" not turn it off completely.

There are two kinds of fuzzy matching that Nominatim does: leaving out whole words and correcting single letters. If you run your own instance, you could play with a Photon version that only does the letter correction. Change minimumShoudMatch to always be 100% in this line: https://github.com/komoot/photon/blob/716b45f969afa22c72f3ab210515c0a5e3e9fee2/src/main/java/de/komoot/photon/elasticsearch/PhotonQueryBuilder.java#L67

**Reply by olivier5741** [reply]
Thanks lonvia, very helpfull !! It seems the issue are the special words such as Av and Avenue. If they could be given much less noise on string comparison, that would make it work ?

**Reply by lonvia** [reply]
Indeed, I'd love to be able to do that but I haven't found an easy way to do that in ElasticSearch. Also note that it is not as simple as degrading any occurrence of 'Avenue' as can be attested by the rather popular name of [Avenue Road](https://nominatim.openstreetmap.org/ui/search.html?q=avenue+road). To get this right, we'd need to be able to give weights to the search terms that are attached to the entry in the database.

**Comment by olivier5741** [comment]
Hi lonvia, is there a way to quickly check photon 0.4.1 result for https://photon.komoot.io/api?q=Avenue%20des%20bons%20l%C3%A9gumes,%201050%20Ixelles,%20Belgique ? I started using photon over nominatim because it give better results for localisation and I have the impression photon might have given me no result for `Avenue des bons légumes` one or two month ago (I might be wrong ...)
